executors/cpp/datetime_fmt.cpp

```diff
@@ -0,0 +1,257 @@
+      cout << "# year: " << year_str << endl;
+    }
+
+    option_item = json_object_object_get(options_obj, "month");
+    if (option_item) {
+      month_str = json_object_get_string(option_item);
+      cout << "# month: " << month_str << endl;
+    }
+
+    option_item = json_object_object_get(options_obj, "weekday");
+    if (option_item) {
+      weekday_str = json_object_get_string(option_item);
+      cout << "# weekdat: " << weekday_str << endl;
+    }
+
+    option_item = json_object_object_get(options_obj, "day");
+    if (option_item) {
+      day_str = json_object_get_string(option_item);
+      cout << "# day: " << day_str << endl;
+    }
+
+    option_item = json_object_object_get(options_obj, "hour");
+    if (option_item) {
+      hour_str = json_object_get_string(option_item);
+      cout << "# hour: " << hour_str << endl;
+    }
+
+    option_item = json_object_object_get(options_obj, "minute");
+    if (option_item) {
+      minute_str = json_object_get_string(option_item);
+      cout << "# minute: " << minute_str << endl;
+    }
+
+    option_item = json_object_object_get(options_obj, "second");
+    if (option_item) {
+      second_str = json_object_get_string(option_item);
+      cout << "# second: " << second_str << endl;
+    }
+
+    option_item = json_object_object_get(options_obj, "fractional_seconds_digits");
+    if (option_item) {
+      fractional_seconds_digits = json_object_get_int(option_item);
+      cout << "# fractional seconds digits: " << fractional_seconds_digits << endl;
+    }
+
+  }
+
+  json_object *date_skeleton_obj = json_object_object_get(json_in, "datetime_skeleton");
+  if (date_skeleton_obj) {
+    // Data specifies a date time skeleton. Make a formatter based on this.
+    string skeleton_string = json_object_get_string(date_skeleton_obj);
+
+    cout << "# Skelecton = " << skeleton_string << endl;
+
+    UnicodeString u_skeleton = skeleton_string.c_str();
+    df = DateFormat::createInstanceForSkeleton(u_skeleton,
+                                               displayLocale,
+                                               status);
+    if (U_FAILURE(status)) {
+      // TODO: Return error.
+      cout << "# ERROR in createInstanceForSkeleton" << endl;
+    }
+  } else {
+    // Create default formatter
+    df = DateFormat::createDateTimeInstance(
+        date_style,
+        time_style,
+        displayLocale);
+  }
+
+  if (df == nullptr) {
+    cout << "# DATE TIME FORMATTER == nullptr == " << df << endl;
+    // Post an error in the return
+    return "";
+  }
+
+  // JSON data returned.
+  json_object *return_json = json_object_new_object();
+  json_object_object_add(return_json, "label", label_obj);
+
+  UDate testDateTime;
+  // The type of conversion requested
+
+  // Prefer milliseconds as input.
+  json_object *input_millis = json_object_object_get(json_in, "input_millis");
+  if (input_millis) {
+    testDateTime = json_object_get_double(input_millis);
+  } else {
```

Comment:
This is a bit messy; you should either have the timestamp or the ISO string but having both just creates confusion. I prefer the ISO string because it is less ambiguous.

Reply:
OK, I'm removing the milliseconds as an output field.

Reply:
Reversal - parsing the date isn't yet working, so I'm keeping the milliseconds for now.

Reply:
Let's chat about how to use `absl` here, which I think is the right thing to do.